src/Build.UnitTests/ChangeWaves_Tests.cs

```diff
@@ -0,0 +1,219 @@
+﻿using System;
+using System.Collections.Generic;
+using Xunit;
+using Shouldly;
+using Microsoft.Build.Utilities;
+using System.Threading;
+using Microsoft.Build.Shared;
+using Microsoft.Build.UnitTests;
+using Xunit.Abstractions;
+using Microsoft.Build.Evaluation;
+
+namespace Microsoft.Build.Engine.UnitTests
+{
+    sealed public class ChangeWaves_Tests
+    {
+        [Theory]
+        [InlineData("16.8")]
+        [InlineData("16.10")]
+        [InlineData("17.0")]
+        [InlineData("25.87")]
+        [InlineData("102.87")]
+        public void EnableAllFeaturesBehindChangeWavesEnablesAllFeaturesBehindChangeWaves(string featureWave)
+        {
+            using (TestEnvironment env = TestEnvironment.Create())
+            {
+                env.SetEnvironmentVariable("MSBUILDDISABLEFEATURESFROMVERSION", ChangeWaves.EnableAllFeatures);
+                ChangeWaves.IsChangeWaveEnabled(featureWave).ShouldBe(true);
```

Comment:
refresh

Reply:
it's outdated at this point, but what did you mean here?

Reply:
I wanted you to call `BuildEnvironmentHelper.ResetInstance_ForUnitTestsOnly();`